main.tf

```
@@ -1,16 +1,18 @@
-provider "aws" {}
+provider "aws" {
+}
 
 locals {
   iam_role_name   = "flow-log-${var.vpc_id}"
-  log_group_name  = "${var.log_group_name == "" ? "/aws/vpc/flow-log/${var.vpc_id}" : var.log_group_name}"
-  iam_role_arn    = "${var.iam_role_arn == "" ? join("", aws_iam_role.this.*.arn) : var.iam_role_arn}"
-  create_iam_role = "${var.log_destination_type == "cloud-watch-logs" && var.iam_role_arn == ""}"
+  log_group_name  = var.log_group_name == null ? "/aws/vpc/flow-log/${var.vpc_id}" : var.log_group_name
```

Does this should work with a `null` default for var.vpc_id?

```suggestion
  log_group_name  = var.log_group_name == null ? "/aws/vpc/flow-log/${tostring(var.vpc_id)}" : var.log_group_name
```

---

no it does not :(